File: src/SalaryCounter.Model/Dto/Employee.cs
using System;
using Newtonsoft.Json;
using SalaryCounter.Model.Attribute;
using SalaryCounter.Model.Util;

namespace SalaryCounter.Model.Dto
{
    /// <summary>
    ///     Employee model
    /// </summary>
    [ModelExample(typeof(EmployeeExample))]
    public sealed class Employee
    {
        ///<inheritdoc cref="Employee"/>
        public Employee(string name, DateTime employmentDate, decimal salaryBase,
            EmployeeType employeeType, Guid? id = null, Guid? chief = null)
        {
            Name = name;
            EmploymentDate = employmentDate;
            SalaryBase = salaryBase;
            EmployeeType = employeeType;
            Id = id ?? Guid.Empty;
            Chief = chief;
        }

        /// <summary>
        ///     Unique employee identifier
        /// </summary>
        [JsonProperty]
        public Guid Id { get; set; }

        /// <summary>
        ///     Employee name
        /// </summary>
        [JsonProperty]
        public string Name { get; set; }

        /// <summary>
        ///     Employment date
        /// </summary>
        [JsonProperty]
        public DateTime EmploymentDate { get; set; }

        /// <summary>
        ///     Base salary
        /// </summary>
        [JsonProperty]
        public decimal SalaryBase { get; set; }

        /// <summary>
        ///     Employee's position
        /// </summary>
        [JsonProperty]
        public EmployeeType EmployeeType { get; set; }

        /// <summary>
        ///     Employee chief's identifier
        /// </summary>
        [JsonProperty]
        public Guid? Chief { get; set; }
    }

    internal class EmployeeExample : Example<Employee>
    {
        protected override Employee Get(IExampleService exampleService) => new(
            "Employee Name", DateTime.Today.AddYears(-1),
            1000, exampleService.Get<EmployeeType>());
    }
}

File: front/angular/src/app/models/employee.ts
import Util from "../util/util";
import {EmployeeType} from "./employeeType";

export class Employee {
    id: string;
    name: string;
    employmentDate: Date;
    salaryBase: number;
    employeeType?: EmployeeType;
    chief?: string;

    constructor() {
        this.id = Util.getEmptyId();
        this.name = "";
        this.salaryBase = 1000;
        this.employmentDate = new Date();
    }

    static fromData(other: any): Employee {
        let employee = new Employee();
        employee.id = other.id || employee.id;
        employee.name = other.name || employee.name;
        employee.salaryBase = other.salaryBase && Number(other.salaryBase) || employee.salaryBase;
        employee.employmentDate =
            other.employmentDate && new Date(other.employmentDate) || employee.employmentDate;
        employee.employeeType =
            other.employeeType && EmployeeType.fromData(other.employeeType) || undefined;
        employee.chief = other.chief || employee.chief;
        return employee;
    }

    static fromDataList(other: any[]): Employee[] {
        return other.map(
            (employee: any, _index: Number, _employees: any[]) =>
                Employee.fromData(employee));
    }
}
